content/tracing/setup/first_class_dimensions.md

```diff
@@ -0,0 +1,149 @@
+
+  * **Go**:
+    ```
+    tracer.SetMeta(“env”, “prod”)
+    ```
+  For OpenTracing use the “GlobalTags” field in the “opentracing.Configuration” structure.
+
+  * **Java**:  
+      Via sysprop: 
+      ```
+      -Ddd.trace.span.tags=”env:prod”
+      ```
+      Via env var:
+      ```
+      DD_TRACE_SPAN_TAGS=”env:prod”
+      ```
+
+  * **Ruby**:
+  ```
+  Datadog.tracer.set_tags(‘env’ => ‘prod’)
+  ```
+
+  * **Python**:
+    ```
+    from ddtrace import tracer
+    tracer.set_tags('env', 'prod')
+    ```
+
+### Viewing Data by Environment
+
+Environments appear near the top of APM pages. Use the dropdown to scope the data displayed on the current page.
+
+{{< img src="tracing/setup/first_class_dimensions/envs_tracing_screen.png" alt="Envs tracing" responsive="true" popup="true" style="width:80%;">}}
+
+
+## Primary Tags
+### Definition
+
+A primary tag is a first-class dimension that you use to scope an entire Datadog APM application. Primary tags can be used in conjunction with environments to get an even finer view of your application’s behavior. Use cases for primary tags include:
+
+* availability zone
+* datacenter
+
+You can optionally set 1 primary tag. A primary tag must follow these rules:
+
+* It must start with a lowercase letter.
+* Other characters must be alphanumeric lowercase Unicode characters, underscores, minuses, colons, periods, or slashes.
+* It must not be more than 100 characters long.
+
+### Setup
+APM primary tags must be set up in two ways: in the trace Agent and in the Datadog UI.
+
+#### Agent configuration
+Override the default tag used by the trace Agent in the Agent configuration file. This tags all traces coming through the Agent, overriding the host tag value. An APM primary tag is configured like any regular host tag. You can specify it as:
+
+```
+tags:
+- key:value
+```
+
+#### Datadog UI
+
+Visit the [APM Settings](https://app.datadoghq.com/apm/settings) page to define, change, or remove primary tags. Only organization admins have access to this page. 
+
+Changes will be reflected in the UI within several minutes.  Note:
+
+ If you change a previously set primary tag, please be aware of the following:
+
+* Historical APM data aggregated by the previously set tag will no longer be accessible.
+* Any APM monitors scoped to the previous tag will display a status of _No Data_.
+
+### Viewing Data by Primary Tag
+
+Primary tags appear near the top of APM pages, next to environments. Use these selectors to slice the data displayed on the current page. To view all data independent of a primary tag, choose `tag-name:*` from the dropdown (as in the image below).
+
+{{< img src="tracing/setup/first_class_dimensions/primary_tags_ui.png" alt="Primary tags UI" responsive="true" popup="true" style="width:80%;">}}
+
+## Further Reading
+
+{{< partial name="whats-next/whats-next.html" >}}
+
+[1]: /agent/tagging
+[2]: /agent/faq/where-is-the-configuration-file-for-the-agent
```

Comment:
@l0k0ms I think this link no longer works.  How about:
https://docs.datadoghq.com/agent/basic_agent_usage/#configuration-file